bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -1731,3 +1734,48 @@ impl TryFrom<SlidingSyncVersion> for SdkSlidingSyncVersion {
         })
     }
 }
+
+#[derive(uniffi::Enum)]
+pub enum OidcPrompt {
+    /// The Authorization Server must not display any authentication or consent
+    /// user interface pages.
+    None,
+
+    /// The Authorization Server should prompt the End-User for
+    /// reauthentication.
+    Login,
+
+    /// The Authorization Server should prompt the End-User for consent before
+    /// returning information to the Client.
+    Consent,
+
+    /// The Authorization Server should prompt the End-User to select a user
+    /// account.
+    ///
+    /// This enables an End-User who has multiple accounts at the Authorization
+    /// Server to select amongst the multiple accounts that they might have
+    /// current sessions for.
+    SelectAccount,
+
+    /// The Authorization Server should prompt the End-User to create a user
+    /// account.
+    ///
+    /// Defined in [Initiating User Registration via OpenID Connect](https://openid.net/specs/openid-connect-prompt-create-1_0.html).
+    Create,
+
+    /// An unknown value.
+    Unknown { value: String },
+}
+
+impl From<OidcPrompt> for SdkOidcPrompt {
```

Comment:
Consider annotating the `SdkOidcPrompt` with the uniffi derive in place of adding the wrapper, since the wrapper seems completely equivalent?

Reply:
`Prompt` comes from [an external crate](https://docs.rs/oauth2-types/0.11.0/oauth2_types/requests/enum.Prompt.html), so I think there's no way to export it? Also, the enum uses the `Case(value)` form, which is not allowed by uniffi 🫤 .

Reply:
It's true the name makes you think the type comes from the SDK... I can rename it to something better in another PR, but nothing came to mind. `OAuth2OidcPrompt`? Should I just use the full `oauth2types::requests::Prompt` form?

Reply:
Do not fret too much about it, sorry, this was more a light remark at the existing code. But yeah using the fully-qualified type would look great here!